internal/provider/resources/resource_api_token_test.go

```diff
@@ -0,0 +1,568 @@
+				),
+			},
+			// Change properties and check they have been updated in terraform state
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + apiToken(apiTokenInput{
+					Name:        apiTokenName,
+					Description: "new description",
+					Type:        "ORGANIZATION",
+					Roles: []apiTokenRole{
+						{
+							Role:       "ORGANIZATION_OWNER",
+							EntityId:   organizationId,
+							EntityType: "ORGANIZATION",
+						},
+						{
+							Role:       "WORKSPACE_OWNER",
+							EntityId:   workspaceId,
+							EntityType: "WORKSPACE",
+						},
+						{
+							Role:       "DEPLOYMENT_ADMIN",
+							EntityId:   deploymentId,
+							EntityType: "DEPLOYMENT",
+						},
+					},
+					ExpiryPeriodInDays: 30,
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceVar, "description", "new description"),
+					// Check via API that organization api token exists
+					testAccCheckApiTokenExistence(t, checkApiTokensExistenceInput{name: apiTokenName, organization: true, shouldExist: true}),
+				),
+			},
+			// Change the resource type and remove roles
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + apiToken(apiTokenInput{
+					Name:        apiTokenName,
+					Description: utils.TestResourceDescription,
+					Type:        "WORKSPACE",
+					Roles: []apiTokenRole{
+						{
+							Role:       "WORKSPACE_OWNER",
+							EntityId:   workspaceId,
+							EntityType: "WORKSPACE",
+						},
+					},
+					ExpiryPeriodInDays: 30,
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceVar, "type", "WORKSPACE"),
+					resource.TestCheckResourceAttr(resourceVar, "roles.#", "1"),
+					resource.TestCheckResourceAttr(resourceVar, "roles.0.entity_id", workspaceId),
+					resource.TestCheckResourceAttr(resourceVar, "roles.0.entity_type", "WORKSPACE"),
+					resource.TestCheckResourceAttr(resourceVar, "roles.0.role", "WORKSPACE_OWNER"),
+					// Check via API that api token was destroyed and recreated
+					testAccCheckApiTokenExistence(t, checkApiTokensExistenceInput{name: apiTokenName, workspace: true, shouldExist: true}),
+				),
+			},
+			// Change resource type back to ORGANIZATION
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + apiToken(apiTokenInput{
+					Name:        apiTokenName,
+					Description: utils.TestResourceDescription,
+					Type:        "ORGANIZATION",
+					Roles: []apiTokenRole{
+						{
+							Role:       "ORGANIZATION_OWNER",
+							EntityId:   organizationId,
+							EntityType: "ORGANIZATION",
+						},
+						{
+							Role:       "WORKSPACE_OWNER",
+							EntityId:   workspaceId,
+							EntityType: "WORKSPACE",
+						},
+						{
+							Role:       "DEPLOYMENT_ADMIN",
+							EntityId:   deploymentId,
+							EntityType: "DEPLOYMENT",
+						},
+					},
+					ExpiryPeriodInDays: 30,
```

Comment:
```suggestion
```

Reply:
`ExpiryPeriodInDays` cannot be removed after a value has been set previous - this aligns with the API as you can't update that field

tested in acc tests and you get:
```
        Invalid field values: field tokenExpiryPeriodInDays failed check on length;
        tokenExpiryPeriodInDays should be min: 1;, status: 400, requestId:
        bea9e0d6-7aa6-4cf8-b158-78cbb6316b80
```

Reply:
we should make this `ReplaceIfConfigured` as well then and mention that in the description